src/store/sqlite/statements.rs

```diff
@@ -0,0 +1,311 @@
+                        interface_major: row.get(4)?,
+                        ownership: row.get(5)?,
+                    })
+                })
+                .map_err(SqliteError::Query)?
+                .map(|e| e.map_err(SqliteError::Query).and_then(StoredProp::try_from))
+                .collect::<Result<Vec<StoredProp>, SqliteError>>()?;
+
+            Ok(v)
+        })
+    }
+
+    pub(super) fn props_with_ownership(
+        &self,
+        ownership: Ownership,
+    ) -> Result<Vec<StoredProp>, SqliteError> {
+        let ownership_par = RecordOwnership::from(ownership) as u8;
+
+        wrap_sync_call(|| {
+            let mut statement = self
+                .prepare_cached(include_query!(
+                    "queries/properties/read/props_where_ownership.sql"
+                ))
+                .map_err(SqliteError::Prepare)?;
+
+            let v = statement
+                .query_map([ownership_par], |row| {
+                    Ok(StoredRecord {
+                        interface: row.get(0)?,
+                        path: row.get(1)?,
+                        value: row.get(2)?,
+                        stored_type: row.get(3)?,
+                        interface_major: row.get(4)?,
+                        ownership: row.get(5)?,
+                    })
+                })
+                .map_err(SqliteError::Query)?
+                .map(|e| {
+                    e.map_err(SqliteError::Query).and_then(|record| {
+                        let prop = StoredProp::try_from(record)?;
+
+                        debug_assert_eq!(prop.ownership, ownership);
+
+                        Ok(prop)
+                    })
+                })
+                .collect::<Result<Vec<StoredProp>, SqliteError>>()?;
+
+            Ok(v)
+        })
+    }
+
+    pub(super) fn interface_props(&self, interface: &str) -> Result<Vec<StoredProp>, SqliteError> {
+        wrap_sync_call(|| {
+            let mut statement = self
+                .prepare_cached(include_query!(
+                    "queries/properties/read/interface_props.sql"
+                ))
+                .map_err(SqliteError::Prepare)?;
+
+            let v = statement
+                .query_map([interface], |row| {
+                    Ok(StoredRecord {
+                        interface: row.get(0)?,
+                        path: row.get(1)?,
+                        value: row.get(2)?,
+                        stored_type: row.get(3)?,
+                        interface_major: row.get(4)?,
+                        ownership: row.get(5)?,
+                    })
+                })
+                .map_err(SqliteError::Query)?
+                .map(|e| e.map_err(SqliteError::Query).and_then(StoredProp::try_from))
+                .collect::<Result<Vec<StoredProp>, SqliteError>>()?;
+
+            Ok(v)
+        })
+    }
+}
+
+impl Deref for ReadConnection {
+    type Target = Connection;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
```

Comment:
What do you think about adding some test to this module?

Reply:
This file is fairly covered by our "store tests" as you can see:

- Here is the invocation https://github.com/joshuachp/astarte-device-sdk-rust/blob/cfc15ad535a20ae60ff6e56085af5a24a88048be/src/store/sqlite/mod.rs#L540
- And the tests are here https://github.com/joshuachp/astarte-device-sdk-rust/blob/a57661a93acd8d7f4485e72ffaf8bf368f26daf5/src/store/mod.rs#L181

The resulting coverage is pretty high (nearly 90%) https://app.codecov.io/gh/astarte-platform/astarte-device-sdk-rust/pull/367/blob/src/store/sqlite/statements.rs#src/store/sqlite/statements.rs.

Most of the missing lines are for query errors. Those are harder to test since the query are correct.

Should I port some of the test from the store to this module just in case?

Reply:
If it is covered by `store tests` it is ok, we no need to add any other tests.